miso-web/src/main/webapp/pages/editSequencerReference.jsp

```diff
@@ -67,23 +86,292 @@
+                <form:option value="0">(choose)</form:option>
+                <form:options items="${otherSequencerReferences}" itemLabel="name" itemValue="id"/>
+              </form:select>
+            </c:if>
+            <span id="upgradedSequencerReferenceLink"></span>
           </td>
         </tr>
       </table>
+      <c:choose>
+        <c:when test="${fn:contains(SPRING_SECURITY_CONTEXT.authentication.principal.authorities,'ROLE_ADMIN')}">
+          <script type="text/javascript">
+            jQuery(document).ready(function() {
+	          Sequencer.ui.showStatusRows();
+	        });
+          </script>
+        </c:when>
+        <c:otherwise>
+          <script type="text/javascript">
+            jQuery(document).ready(function() {
+              Sequencer.ui.hideStatusRowsReadOnly(
+                  ${sequencerReference.dateDecommissioned != null}, 
+                  ${sequencerReference.upgradedSequencerReference != null}, 
+                  ${sequencerReference.upgradedSequencerReference != null ? sequencerReference.upgradedSequencerReference.id : 0}, 
+                  "<c:if test="${sequencerReference.upgradedSequencerReference != null}">${sequencerReference.upgradedSequencerReference.name}</c:if>"
+              );
+            });
+          </script>
+        </c:otherwise>
+      </c:choose>
       <br/>
     </form:form>
+    
+    <script type="text/javascript">
+      jQuery(document).ready(function () {
+        // Attaches a Parsley form validator.
+        Validate.attachParsley('#sequencer_reference_form');
+      });
+    </script>
+    
+    <div class="sectionDivider" onclick="Utils.ui.toggleLeftInfo(jQuery('#records_arrowclick'), 'recordsdiv');">
+      <c:choose>
+        <c:when test="${fn:length(sequencerServiceRecords) == 1}">1 Service Record</c:when>
+        <c:otherwise>${fn:length(sequencerServiceRecords)} Service Records</c:otherwise>
+      </c:choose>
+      <div id="records_arrowclick" class="toggleLeft"></div>
+    </div>
+    <div id="recordsdiv" style="display:none;">
+      <h1>Service Records</h1>
+      <ul class="sddm">
+        <li>
+          <a onmouseover="mopen('recordmenu')" onmouseout="mclosetime()">
+            Options
+            <span style="float:right" class="ui-icon ui-icon-triangle-1-s"></span>
+          </a>
+          <div id="recordmenu" onmouseover="mcancelclosetime()" onmouseout="mclosetime()">
+            <c:choose>
+              <c:when test="${sequencerReference.dateDecommissioned == null}">
+                <a href='<c:url value="/miso/stats/sequencer/servicerecord/new/${sequencerReference.id}"/> '>Add new Service Record</a>
+              </c:when>
+              <c:otherwise>
+                <a onclick="alert('Error: Cannot add service records to a retired sequencer')">Add new Service Record</a>
```

Comment:
I like having an indication of **why** records can't be added, rather than have the user wonder why the button is gone. I suppose a message could be placed somewhere visible without clicking anything, but I'm not sure I like that better. Thoughts?